test/main.ml

```diff
@@ -0,0 +1,228 @@
+open Monolith
+
+module type Testable = sig
+  type hash = string
+  type commit_metadata = { date : int64; author : string; message : string }
+  type commit = { tree : hash; parents : hash list; metadata : commit_metadata }
+  type entry_kind = Content | Node
+  type tree_entry = { name : string; kind : entry_kind; hash : hash }
```

Comment:
Any reason for defining these types here rather than above? Might save a bit of aliasing boilerplate below.

Reply:
Moving them right above isn't enough, but feel free to push your patch! :+1:

Reply:
Fixed in a new commit.